src/ModelContextProtocol/Server/StreamableHttpServerTransport.cs

```diff
@@ -46,15 +46,18 @@ public sealed class StreamableHttpServerTransport : ITransport
     public bool Stateless { get; init; }
 
     /// <summary>
-    /// Gets the initialize request if it was received by <see cref="HandlePostRequest(IDuplexPipe, CancellationToken)"/> and <see cref="Stateless"/> is set to <see langword="true"/>.
+    /// Gets or sets a callback to be invoked before handling the initialize request.
     /// </summary>
-    public InitializeRequestParams? InitializeRequest { get; internal set; }
+    public Action<InitializeRequestParams>? OnInitRequestReceived { get; set; }
```

Comment:
It feels like this could be generalized (async callback? passing additional context?) but for now I went for the simplest solution possible for the task at hand.

Reply:
@halter73 I converted this to a `Func<InitializeRequestParams, ValueTask>` for the sake of minimizing the probability of future amendments.